### ISSUE TITLE
incorrect indentation for multi-variable declaration

### ISSUE THREAD

**Issue by drichardson** [pangloss/vim-javascript#322]
As of 1d8c2677, indenting the following code

    var a=1,
        b=2,
        c=3

results in the 3rd line being indented an extra level.

    var a=1,
        b=2,
            c=3

This appears to be a regression introduced in 487e0c659. In the previous commit (c3787301c), it indents as expected.

**Comment by MobiusHorizons** [comment]
I can confirm that this happens.

**Comment by sboehler** [comment]
Indentation of multi-variable declarations is indeed broken. I get something like this with current master (1d8c267):

```
var a = 2,
    b = 2
    c = 4;

    function bar() {
    }
    a = 4;
```
While the first var statement is correct, the indentation is not reset after the semicolon.

**Comment by sboehler** [comment]
+1
Also the indentation is not reset after the extra level.
```
var a = 2,
	b = 2,
		c = 4;

		function bar() {

		}
		a = 4;
```

**Comment by Pompeu** [comment]
i have  similar bug
![vim-const-var](https://cloud.githubusercontent.com/assets/7594343/12220377/49a3cb88-b757-11e5-877b-64ad80bcc9ee.png)

**Comment by davidchambers** [comment]
I don't have a solution, but I can suggest a workaround:

```javascript
const assert = require('assert');
const fs     = require('fs');
const net    = require('net');
const jade   = require('jade');
const wrk    = require('../workers/wrk-parser');
```

This style is objectively better since it results in cleaner diffs when imports are added or removed. :)

**Comment by MobiusHorizons** [comment]
^ BUMP

**Comment by davidchambers** [comment]
There's not much point bumping this issue. None of the maintainers if familiar with the indentation code. We'll welcome a contribution from someone who's willing to get stuck in to figure out how it works and fix some of the [issues][1]. :)


[1]: https://github.com/pangloss/vim-javascript/issues?q=is%3Aopen+is%3Aissue+label%3Aindentation

**Comment by robjens** [comment]
I can't seem to duplicate this behavior. To me it actually does indent as it is supposed to, both with lead comma and eol.

```
const foo = 'bar',
    bar = 'baz',
    baz = 'boozle'; <RET>
_ <-- cursor ends here as it is supposed to at col 0
```

I just noticed something odd though, this does *not* happen with `let` keyword usage:

```
let foo = 'baloo',
_ <-- cursor ends at col 0 which is INCORRECT
```

I may take a closer look although `vimL`... meh *shrugs* rather learn emacs lol.

**Comment by amadeus** [comment]
This issue should be resolved in `develop`

**Comment by drichardson** [comment]
@amadeus yup, verified fixed in c4725ca4d95502054e7159c
